File: src/types/workOrder.ts
export type WorkOrderStatus = 
  | "DRAFT" 
  | "OPEN" 
  | "ASSIGNED" 
  | "IN_PROGRESS" 
  | "COMPLETED" 
  | "CANCELLED";

export interface WorkOrder {
  id: string;
  workOrderNumber: string;
  title: string;
  description: string;
  companyId: string;
  vendorId?: string;
  status: WorkOrderStatus;
  assignedDate?: string;
  dueDate?: string;
  completedDate?: string;
  createdAt: string;
  updatedAt: string;
}

export const WORK_ORDER_STATUS_LABELS: Record<WorkOrderStatus, string> = {
  DRAFT: "Draft",
  OPEN: "Open",
  ASSIGNED: "Assigned",
  IN_PROGRESS: "In Progress",
  COMPLETED: "Completed",
  CANCELLED: "Cancelled",
};

export const WORK_ORDER_STATUS_VARIANTS: Record<WorkOrderStatus, "default" | "secondary" | "destructive" | "outline"> = {
  DRAFT: "outline",
  OPEN: "secondary",
  ASSIGNED: "default",
  IN_PROGRESS: "default",
  COMPLETED: "secondary",
  CANCELLED: "destructive",
};

File: backend/src/main/java/com/hourglass/jobclocksync/model/WorkOrder.java
package com.hourglass.jobclocksync.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "workorders")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class WorkOrder {
    @Id
    private String id;
    
    @Indexed(unique = true)
    private String workOrderNumber;
    
    private String title;
    private String description;
    private String companyId;
    private String vendorId;
    private WorkOrderStatus status;
    private LocalDateTime assignedDate;
    private LocalDateTime dueDate;
    private LocalDateTime completedDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    public enum WorkOrderStatus {
        DRAFT, OPEN, ASSIGNED, IN_PROGRESS, COMPLETED, CANCELLED
    }
}

